hugr-model/capnp/hugr-v0.capnp

```diff
@@ -97,13 +101,22 @@ struct Operation {
 
 struct Region {
     kind @0 :RegionKind;
-    sources @1 :List(LinkRef);
-    targets @2 :List(LinkRef);
+    sources @1 :List(LinkIndex);
+    targets @2 :List(LinkIndex);
     children @3 :List(NodeId);
     meta @4 :List(MetaItem);
     signature @5 :OptionalTermId;
+    scope @6 :RegionScope;
+}
+
+struct RegionScope {
+    links @0 :UInt32;
+    ports @1 :UInt32;
 }
 
+# Either `0` for an open scope, or the number of links in the closed scope incremented by `1`.
```

Comment:
Links don't have a global index space, but every closed region defines an index space. There can't be non-local edges across closed regions. This is a first step to simplify upcoming work on consts and nested hugrs, so that ideally closed regions can be loaded as `Hugr`s themselves.